### ISSUE TITLE
Passing arguments to syntax highlighter

### ISSUE THREAD

**Issue by ramsey** [asciidoctor/asciidoctor#1726]
Is there a way to pass arguments to the syntax highlighter? I my case, I'm using Pygments, and I would like to pass additional arguments. In standard asciidoc, it appears [there is a way to pass `args`](http://asciidoc.org/source-highlight-filter.html#_block_attributes), but it doesn't work with asciidoctor (I haven't tested it with asciidoc, either).

Each Pygments lexer has standard options and options specific to the lexer (see here: http://pygments.org/docs/lexers/). If I am using PHP, for example, and wish to highlight the code without including the opening tag (`<?php`), I can pass `startinline=True` to Pygments, and it will highlight the code as if the opening tag were present.

With asciidoctor, it would be nice to allow passing arguments like this:

```
= My Document
:doctype: article
:listing-caption: Listing
:source-highlighter: pygments
:pygments-style: manni
:pygments-linenums-mode: table

.Factorials
[source, php, numbered, args="startinline=True"]
----
function factorial($number)
{
    if ($number < 2) {
        return 1;
    } else {
        return ($number * factorial($number-1));
    }
}
----
```

Is something like this already possible? If not, does this sound like a good addition to asciidoctor?

Thanks!

**Comment by mojavelinux** [comment]
What you are describing is not currently possible. We could add it, though. We should support both a local and a global attribute. I'd prefer if the name were "pygments-args". The value should be a JSON-style hash, which we can parse and pass through to the processor safetly.

```
:pygments-options: startinline:true, funcnamehighlighting:true
```

We may even want to break it down by language:

```
:pygments-php-options: startinline:true
```

On the block, it would be:

```
[source,php,pygments-options="startinline:true"]
```

**Comment by ramsey** [comment]
That sounds good to me. Thanks for considering it.

**Comment by ricpelo** [comment]
PR #2015 solves the issue for `startinline = true`.

**Comment by mojavelinux** [comment]
This will be resolved as of 1.5.6 in the form we discussed.
